### PR TITLE
feat: create journeyTemplate - makes a journey a template or not

### PR DESCRIPTION
**Pull request by GeronimoJohn**
# Description

This updates template

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/28496234/todos/5142646864)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Api-journeys tests to pass
- [x] Api only

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/journey/journey.resolver.ts [hunk 1]
```diff
@@ -400,6 +403,20 @@ export class JourneyResolver {
...
+  @Mutation()
+  async journeyTemplate(
+    @Args('id') id: string,
+    @Args('input') input: JourneyTemplateInput
+  ): Promise<Journey> {
+    const journey: Journey = await this.journeyService.update(id, input)
+    const userJourneys: UserJourney[] = await this.userJourneyService.getAll()
+    const userJourneyIds: string[] = userJourneys.map((user) => user.id)
+    userJourneys != null &&
+      (await this.userJourneyService.removeAll(userJourneyIds))
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/journey/journey.resolver.ts:415]: @storyworks let me know if this is where you wanted me to remove userJourneys

Comment (storyworks) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:415]: close! The userJourney call should be separately done in the front end. So not needed here
image

**Comment by csiyang** [comment]
This mutation still needs the updated role guard to be implemented

**Comment by csiyang** [comment]
I tested this mutation on the staging environment

**Comment by storyworks** [comment]
Just awaiting role guard

File: apps/api-journeys/src/app/modules/journey/journey.resolver.spec.ts [hunk 2]
```diff
@@ -882,6 +889,13 @@ describe('JourneyResolver', () => {
     })
   })
 
+  describe('journeyTemplate', () => {
+    it('updates template and removes userJourney', async () => {
```
Comment (storyworks) [inline apps/api-journeys/src/app/modules/journey/journey.resolver.spec.ts:893]: `updates template`

File: libs/nest/database/src/lib/BaseService.ts [hunk 3]
```diff
@@ -84,6 +84,14 @@ export abstract class BaseService {
     return result.old
   }
 
+  @KeyAsId()
+  async removeAll<T>(keys: string[]): Promise<T[]> {
```
Comment (storyworks) [inline libs/nest/database/src/lib/BaseService.ts:88]: Am assuming this change will disappear when you merge in the other PR.

Comment (csiyang) [reply libs/nest/database/src/lib/BaseService.ts:88]: @storyworks No, this needs to stay as it is used by the resolver...

Comment (storyworks) [reply libs/nest/database/src/lib/BaseService.ts:88]: @csiyang , it should be part of the other PR, not this one which has nothing to do with removing anything. Once you merge the other PR, this code will disappear

**Review by storyworks — APPROVED** [review]

**Pull request merged**